api/src/s2-api.ts

```diff
@@ -48,23 +48,32 @@ async function getPaper(s2Id: string): Promise<Paper | undefined> {
     console.error("Failed to fetch data from Semantic Scholar API", err);
     return;
   }
+
   if (isS2ApiResponseSuccess(response)) {
-    const data = response.data as S2ApiPaper;
-    const year = parseInt(data.year) || null;
+    const data = response.data as S2ApiPaperResponse;
+    if (data.paper === undefined) {
+      return;
+    }
+    var year;
+    if (data.paper.year.text === undefined) {
+      year = null;
+    } else {
+      year = parseInt(data.paper.year.text)
+    }
     return {
       s2Id,
-      title: data.title,
-      authors: data.authors.map(a => ({
-        id: a.authorId,
+      title: data.paper.title.text,
+      authors: data.paper.authors[0].map(a => ({
+        id: a.ids,
         name: a.name,
-        url: a.url
       })),
-      abstract: data.abstract,
-      url: data.url,
+      abstract: data.paper.paperAbstract.text,
       year,
-      venue: data.venue,
-      citationVelocity: data.citationVelocity || 0,
-      influentialCitationCount: data.influentialCitationCount || 0
+      venue: data.paper.venue.text,
+      citationVelocity: data.paper.citationStats.citationVelocity || 0,
+      influentialCitationCount: data.paper.citationStats.numKeyCitations || 0,
+      primaryPaperLink: data.paper.primaryPaperLink.url,
+      paperLinkType: data.paper.primaryPaperLink.linkType,
```

Comment:
nit pick: can this be `primaryPaperLinkType` to make it clear that it is related to `primaryPaperLink`?